src/util-file-decompression.c

```diff
@@ -0,0 +1,191 @@
+/* Copyright (C) 2015 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/** \file
+ *
+ * \author Giuseppe Longo <giuseppe@glongo.it>
+ *
+ * \brief Decompress files transfered via HTTP corresponding to file_data
+ * keyword.
+ *
+ */
+
+#include "suricata-common.h"
+#include "suricata.h"
+
+#include "app-layer-htp.h"
+
+#include "util-file-decompression.h"
+#include "util-file-flash-decompression.h"
+#include "util-misc.h"
+#include "util-print.h"
+
+#define FLASH_ZLIB_MIN_VERSION    0x06
+#define FLASH_LZMA_MIN_VERSION    0x0D
+
+int FileIsFlashFile(const uint8_t *buffer, uint32_t buffer_len)
+{
+    if (buffer_len >= 3 && buffer[1] == 'W' && buffer[2] == 'S') {
+        if (buffer[0] == 'F')
+            return FILE_FLASH_NO_COMPRESSION;
+        else if (buffer[0] == 'C')
+            return FILE_FLASH_ZLIB_COMPRESSION;
+        else if (buffer[0] == 'Z')
+            return FILE_FLASH_LZMA_COMPRESSION;
+        else
+            return FILE_IS_NOT_FLASH;
+    }
+
+    return FILE_IS_NOT_FLASH;
+}
+
+int FileDecompressFlashFile(const uint8_t *buffer, uint32_t *buffer_len,
```

Comment:
I think it's cleaner to have uint32_t buffer_len (so not a pointer to return data back to the caller), and instead add a uint32_t *decompressed_buffer_len to pass back the size of uint8_t **decompressed_buffer